conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegration.java

```diff
@@ -95,6 +97,22 @@ public ConvivaAnalyticsIntegration(Player player,
         }
 
         attachBitmovinEventListeners();
+        setUpAdAnalyticsCallback();
+    }
+
+    private void setUpAdAnalyticsCallback() {
+        convivaAdAnalytics.setCallback(new ConvivaExperienceAnalytics.ICallback() {
+            @Override
+            public void update() {
+                if (bitmovinPlayer.isAd()) {
+                    convivaAdAnalytics.reportAdMetric(ConvivaSdkConstants.PLAYBACK.PLAY_HEAD_TIME, ((long) (bitmovinPlayer.getCurrentTime() * 1000)));
+                }
+            }
+
+            @Override
+            public void update(String s) {
+            }
+        });
```

Comment:
Playbacktime of the main content is done in the `TimeChanged` event listener.
Couldn't we reuse the same code path for ads as well?

What's the advantage of this callback?

Reply:
We could do the same here, but the callback is the suggested way of reporting buffer levels, playhead time and rendered frames. 
In the future we should also change over the main content reporting to the callback style to avoid over reporting.